cvat-sdk/cvat_sdk/core/proxies/model_proxy.py

```diff
@@ -210,3 +217,128 @@ def remove(self: Entity) -> None:
         """
 
         self.api.destroy(id=getattr(self, self._model_id_field))
+
+
+class _ExportMixin(Generic[_EntityT]):
+    def export(
+        self,
+        endpoint: Callable,
+        filename: StrPath,
+        *,
+        pbar: Optional[ProgressReporter] = None,
+        status_check_period: Optional[int] = None,
+        location: Optional[Location] = None,
+        cloud_storage_id: Optional[int] = None,
+        **query_params,
+    ) -> None:
+        query_params = {
+            **query_params,
+            **({"location": location} if location else {}),
+        }
+
+        if location == Location.CLOUD_STORAGE:
+            if not cloud_storage_id:
+                raise ValueError(
+                    f"Cloud storage ID must be specified when {location!r} location is used"
+                )
+
+            query_params["cloud_storage_id"] = cloud_storage_id
+
+        local_downloading = (
+            location == Location.LOCAL
+            or not location
+            and (not self.target_storage or self.target_storage.location.value == Location.LOCAL)
+        )
+
+        if not local_downloading:
+            query_params["filename"] = str(filename)
+
+        downloader = Downloader(self._client)
+        bg_request = downloader.prepare_file(
+            endpoint,
+            url_params={"id": self.id},
+            query_params=query_params,
+            status_check_period=status_check_period,
+        )
+
+        result_url = bg_request.result_url
+        if local_downloading:
+            assert result_url
+        else:
+            assert not result_url
+
+        if result_url:
+            downloader.download_file(result_url, output_path=Path(filename), pbar=pbar)
```

Comment:
Probably, the asserts should be changed to a different error, as such situations mean the fetched model was outdated.